fixieai/agents-ts/src/__tests__/serve.ts

```diff
@@ -3,14 +3,58 @@ import got from 'got';
 import nock from 'nock';
 import path from 'path';
 import tempy from 'tempy';
+import { Jsonifiable } from 'type-fest';
 import type { PromiseType } from 'utility-types';
 import serve from '../serve';
 
+/* eslint-disable id-blacklist */
+
+const agentId = 'my-agent';
 const agentPackagePath = path.resolve(__dirname, '..', 'fixtures', 'normal');
 let refreshMetadataAPIUrlCallCount = 0;
 const refreshMetadataAPIUrl = 'http://fake:3000/refresh-metadata';
+const userStorageApiUrl = 'http://fake:3000/user-storage';
+
+// TODO: make a `got` client to reduce duplication.
+
+const gotClient = got.extend({
+  hooks: {
+    beforeError: [
+      (error) => {
+        console.log(error);
+        console.log(error.response?.body);
+        return error;
+      },
+    ],
+  },
+});
+
+const mockDataStore: Record<string, Jsonifiable> = {};
+nock(new URL(userStorageApiUrl).origin, {
+  reqheaders: {
+    authorization: 'Bearer auth-token',
+  },
+}).persist()
+  .get(new RegExp(`/user-storage/${agentId}/`)).reply(200, (uri) => {
+    const key = uri.split('/').pop();
+    return {data: mockDataStore[key!]};
+  })
+  .get(new RegExp(`/user-storage/${agentId}`)).reply(200, () => Object.keys(mockDataStore).map(key => ({key})))
+  .post(new RegExp(`/user-storage/${agentId}/`)).reply(200, (uri, requestBody) => {
+    const key = uri.split('/').pop();
+    mockDataStore[key!] = (requestBody as Record<string, any>).data;
+  })
+  .head(new RegExp(`/user-storage/${agentId}/`)).reply((uri) => {
+    const key = uri.split('/').pop();
+    return [key !== undefined && key in mockDataStore ? 200 : 404];
+  })
+  .delete(new RegExp(`/user-storage/${agentId}/`)).reply(200, (uri) => {
+    const key = uri.split('/').pop();
+    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+    delete mockDataStore[key!];
+  });
 
-nock(new URL(refreshMetadataAPIUrl).origin).post('/refresh-metadata').times(Infinity).reply(
+nock(new URL(refreshMetadataAPIUrl).origin).persist().post('/refresh-metadata').reply(
```

Comment:
`persist()` is a cleaner way to express what `times(Infinity)` was trying to do.